Tests/GitHubRepositories.tests.ps1

```diff
@@ -290,6 +290,72 @@ try
         }
     }
 
+    Describe 'GitHubRepositories\New-GitHubRepositoryFromTemplate' {
+        BeforeAll -ScriptBlock {
+            $templateRepoName = ([Guid]::NewGuid().Guid)
+            $testGitIgnoreTemplate=(Get-GitHubGitIgnore)[0]
+            $testLicenseTemplate=(Get-GitHubLicense)[0].key
+
+            $newGitHubRepositoryParms = @{
+                RepositoryName = $templateRepoName
+                Description = $defaultRepoDesc
+                GitIgnoreTemplate = $testGitIgnoreTemplate
+                LicenseTemplate = $testLicenseTemplate
+                IsTemplate = $true
+            }
+            $templateRepo = New-GitHubRepository @newGitHubRepositoryParms
+        }
+
+        Context -Name 'When creating a public repository from a template' -Fixture {
+            BeforeAll -ScriptBlock {
+                $repoName = ([Guid]::NewGuid().Guid)
+                $newRepoDesc = 'New Repo Description'
+                $newGitHubRepositoryFromTemplateParms = @{
+                    RepositoryName = $repoName
+                    OwnerName = $script:ownerName
+                    TemplateOwnerName = $templateRepo.owner.login
+                    TemplateRepositoryName = $templateRepoName
+                    Description = $newRepoDesc
+                }
+                $repo = New-GitHubRepositoryFromTemplate @newGitHubRepositoryFromTemplateParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $repo | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $repo.name | Should -Be $repoName
+                $repo.private | Should -BeFalse
+                $repo.owner.login | Should -Be $script:ownerName
+                $repo.description | Should -Be $newRepoDesc
+                $repo.is_template | Should -BeFalse
+            }
+
+            It 'Should have created a .gitignore file' {
+                { Get-GitHubContent -Uri $repo.svn_url -Path '.gitignore' } | Should -Not -Throw
+            }
+
+            It 'Should have created a LICENSE file' {
+                { Get-GitHubContent -Uri $repo.svn_url -Path 'LICENSE' } | Should -Not -Throw
+            }
+
+            AfterAll -ScriptBlock {
+                if ($repo)
+                {
+                    Remove-GitHubRepository -Uri $repo.svn_url -Confirm:$false
+                }
+            }
+        }
+
+        AfterAll -ScriptBlock {
+            if ($repo)
```

Comment:
```powershell
            if ($templateRepo)
```

Reply:
That being said, it's not clear to me that either of those if checks are necessary.

Reply:
Ah, now that you mention it, I've encountered errors like that before during development of some other features and corresponding tests.  If you want to add the check to avoid the additional error, that's fine.

There's no need to specify `Set-StrictMode -Version 1.0` in any test...that's done for you already in `Tests\Common.ps1` which gets `dot invoked` at the top of every test file, thereby bringing that setting in.

Reply:
OK, I've removed it from the test file.